app/uk/gov/hmrc/cdsreimbursementclaimstubs/models/SubmitClaimResponse.scala

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cdsreimbursementclaimstubs.models
+
+final case class SubmitClaimResponse(
+  response: Either[Either[WAFErrorResponse, EisErrorResponse], EisResponse]
```

Comment:
How about this:

`trait ErrorResponse`
`EisErrorResponse extends ErrorResponse`
`WAFErrorResponse extends ErrorResponse`
`type  SubmitClaimResponse = Either[ErrorResponse, EisResponse]`
But I know it could cause some trouble with the Reads

Reply:
No, it's fine as it is. It's clear. Don't be scared of types!